cdi-core/src/main/java/com/linkedin/cdi/connection/S3Connection.java

```diff
@@ -130,32 +130,44 @@ public WorkUnitStatus execute(WorkUnitStatus status) {
     return status;
   }
 
-  private ByteArrayInputStream handleUpload(String finalPrefix) {
+  /**
+   * Get s3 key either from activation parameters named `s3key` or from the source path itself
+   */
+  @NotNull
+  private String getS3Key(Path path) {
+    JsonObject activationParameters = getExtractorKeys().getActivationParameters();
+    for (Map.Entry<String, JsonElement> entry : activationParameters.entrySet()) {
```

Comment:
Can you use "has" and "getAsJsonPrimitive" method here
https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.5/com/google/gson/JsonObject.html
instead of using the loop?
And make sure that if the custom key doesn't exist return the original path.getName()

Reply:
Good catch. It's already return path.getName() by default, I change to 

@NotNull
  private String getS3Key(Path path) {
    JsonObject activationParameters = getExtractorKeys().getActivationParameters();
    if(activationParameters.has(UPLOAD_S3_KEY)){
      return activationParameters.get(UPLOAD_S3_KEY).getAsString();
    }
    return path.getName();
  }